packages/zowe-explorer/src/trees/shared/SharedUtils.ts

```diff
@@ -333,4 +333,19 @@ export class SharedUtils {
             });
         }
     }
+
+    /**
+     * Debounces an event handler to prevent duplicate triggers.
+     * @param callback Event handler callback
+     * @param delay Number of milliseconds to delay
+     */
+    public static debounce<T extends (...args: any[]) => void>(callback: T, delay: number): (...args: Parameters<T>) => void {
```

Comment:
Should we add patch coverage for this function? I see its being mocked for one of the tests, but we might want to validate the debounce behavior just in case its used elsewhere 😋